admin/class-acc_user_importer-admin.php

```diff
@@ -356,94 +301,58 @@ private function proccess_user_data ( $users ) {
 			$userCity = $user["City"] ?? '';
 			
 			//Log the info we received for this user
-			$user_info = $userContactId;
-			$user_info = $user_info . " " . $userEmail;
-			$user_info = $user_info . " home:" . $userHomePhone;
-			$user_info = $user_info . " cell:" . $userCellPhone;
-			$user_info = $user_info . " member#:" . $userMembership;
-			$user_info = $user_info . " expiry:" . $userExpiry;
-			$this->log_dual("Received " . $user_info);
+			$userInfoString = $userContactId;
+			$userInfoString .= " " . $userEmail;
+			$userInfoString .= " home:" . $userHomePhone;
+			$userInfoString .= " cell:" . $userCellPhone;
+			$userInfoString .= " member#:" . $userMembership;
+			$userInfoString .= " expiry:" . $userExpiry;
+			$this->log_dual("Received " . $userInfoString);
+
+
+			// Skip users without membership number
+			if (!is_numeric($userMembership) || !is_numeric($userContactId)) {
+				$this->log_dual("User is missing membership number or contact ID, skipping.");
+				continue;
+			}
 
 			//Create an array for the core wordpress user information
-			$user_data = array(
+			$accUserData = [
 				'first_name'	=>	$userFirstName,
 				'last_name'		=>	$userLastName,
-				'display_name'	=>	$userContactId,
+				'display_name'	=>	$userFirstName . " " . $userLastName,
 				'user_nicename'	=>	strtolower($userFirstName . "-" . $userLastName),
 				'user_login'	=>	$userContactId,
 				'user_email'	=>	$userEmail,
 				'user_pass'		=>	null,
-			);
+			];
+
+			$accUserMetaData = [
+				'home_phone' => $userHomePhone,
+				'cell_phone' => $userCellPhone,
+				'membership' => $userMembership,
+				'expiry' => $userExpiry,
+				'city' => $userCity
+			];
 			
 			//check if ID already exist
 			$user_id = username_exists($userContactId);
-			$update_this_user = false;
-			
+			$updatedFields = [];
+
+			// Existing user, check if any fields were updated
 			if( is_numeric( $user_id ) ) {
 				//---------USER WAS FOUND IN DATABASE------------
 				$this->log_dual(" > found " . $userContactId . " (user #" . $user_id . ")");
```

Comment:
Here in the above log I would really want to print the user name (first, last) instead of the numeric contactID. I think it would need to be like this
$this->log_dual(" > found " . $accUserData['display_name'] . " (user #" . $user_id . ")");

Reply:
Good catch, I intended to change this but forgot about it.